UT4MasterServer/Controllers/AccountController.cs

```diff
@@ -181,8 +181,22 @@ public async Task<IActionResult> RegisterAccount([FromBody] RegisterRequest requ
 			return Conflict("Username already exists");
 		}
 
-		// TODO: should we also get user's email?
-		await accountService.CreateAccountAsync(request.Username, request.Password); // TODO: this cannot fail?
+		var email = await accountService.GetAccountEmailAsync(request.Email);
+
+		Regex regex = new Regex(@"^(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|""(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*"")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])");
+		Match match = regex.Match(request.Email);
+		if (!match.Success)
+		{
+			logger.LogInformation($"Entered an incorrect email format: {request.Email}");
+			return Conflict("You have entered and invalid email address");
+		}
+		if (email != null)
+		{
+			logger.LogInformation($"Could not register duplicate email: {request.Email}");
+			return Conflict("Email already exists");
+		}

```

Comment:
This should be checked immediately after trying to get email account. so this if should be on line 185.